src/backend/libc/termios/syscalls.rs

```diff
@@ -27,6 +27,27 @@ pub(crate) fn tcgetattr(fd: BorrowedFd<'_>) -> io::Result<Termios> {
     }
 }
 
+#[cfg(all(
+    any(target_os = "android", target_os = "linux"),
+    any(
+        target_arch = "x86",
+        target_arch = "x86_64",
+        target_arch = "x32",
+        target_arch = "riscv64",
+        target_arch = " aarch64",
+        target_arch = " arm",
+        target_arch = " mips",
+        target_arch = " mips64",
```

Comment:
Here and in the rest of the PR, are those spaces in the strings intended?

Reply:
Hot damn, #543 is great!

> make the public Termios type just be struct termios2

Would that be a breaking change? And what about the CFGs? It seems like termios2 isn't a thing outside those architectures, but OG termios is.

Reply:
I meant to say make `Termios` be `termios2` on Linux etc., and `termios` otherwise. It looks like `termios2` is a superset of `termios`. I guess one potentially annoying thing about that approach is if users are deconstructing the struct, it obliges them to name the other fields, but maybe they can just do `..`?

Reply:
Gotya, that seems reasonable :+1:

Reply:
Hmmm, what do we do about libc and its methods like `cfgetospeed`. Do we re-implement them? That doesn't seem portable so maybe libc just doesn't get termios2? Or maybe we just keep 'em separate?

Reply:
Oh, good point. Yeah, I'd like to avoid linux-raw gaining APIs that we can't implement on libc. And on libc we should stick with libc rather than reimplementing these. So I guess we should keep 'em separate.

Reply:
Done.